### PR TITLE
ACM-11084: update hive to mce-2.6

### PR DESCRIPTION
**Pull request by dtthuynh**
https://issues.redhat.com/browse/ACM-11084

### CODE REVIEW

**Comment by dtthuynh** [comment]
/cc @fxiang1 
Started to update the hive client

**Comment by dtthuynh** [comment]
/unhold
/cc @fxiang1 @chenz4027 
FYI @atifshafi , we'll need to make sure we run some additional e2e checks for cluster curator.

File: pkg/jobs/hive/hive_test.go [hunk 1]
```diff
@@ -776,457 +797,457 @@ func TestUpgradeCluster(t *testing.T) {
...
-							}]}}`)
-				client.Patch(context.Background(), &resultmca, clientv1.RawPatch(types.MergePatchType, patch))
-				break
-			}
-		}
-	}()
-
-	assert.Nil(t, UpgradeCluster(client, ClusterName, clustercurator), "Upgrade started successfully to non-recommended version")
-}
+// func TestUpgradeClusterWithChannelUpstream(t *testing.T) {
```
Comment (dtthuynh) [inline pkg/jobs/hive/hive_test.go:800]: @fxiang1 I was able to finally get through everything except for these 4 tests in hive_test.go related to cluster curator upgrade that don't rely on the hive client anyways. If the rest of the PR is okay, would you be okay to rebase your branch with the changes and help take a look at those unit tests?

**Review by chenz4027 — APPROVED** [review]
/lgtm

File: pkg/jobs/hive/hive_test.go (hunk 1 above)
Comment (fxiang1) [reply pkg/jobs/hive/hive_test.go:800]: Sure I can take a look. Thanks David!

**Review by fxiang1 — APPROVED** [review]

**Pull request merged**